openlibrary/plugins/openlibrary/home.py

```diff
@@ -124,12 +125,14 @@ def generic_carousel(query=None, subject=None, work_id=None, _type=None,
     return storify(books) if books else books
 
 @public
-def readonline_carousel():
+def readonline_carousel(query=None):
     """Return template code for books pulled from search engine.
        TODO: If problems, use stock list.
     """
     try:
-        data = random_ebooks()
+        if query in CAROUSELS_PRESETS:
+            query = CAROUSELS_PRESETS[query]
+        data = search_ebooks(query) if query else random_ebooks()
         if len(data) > 30:
             data = lending.add_availability(random.sample(data, 30))
             data = [d for d in data if d['availability']['is_readable']]
```

Comment:
It looks like this is only showing `is_readable` books. Let's see if we can use `QueryCarousel` instead:

https://github.com/internetarchive/openlibrary/blob/82c6fa906b5c2605b9f8352fafcd8f6bd407b905/openlibrary/macros/QueryCarousel.html#L1

You can use it from `home/index.html` like so:

`$:macros.QueryCarousel(query="key:(/works/blah OR ...)")`

Note you'll need to add an option to `QueryCarousel` to set the `url`.